pkg/ingest/conn.go

```diff
@@ -21,6 +22,7 @@ import (
 func HandleConnections(
 	ln net.Listener,
 	h ConnectionHandler,
+	reader func(io.Reader) RecordReader,
```

Comment:
If you take the Factory suggestion, this would become `rfac ingest.RecordReaderFactory`